### PR TITLE
feat(caip-222): multi-chain wallet authentication

### PR DESCRIPTION
**Pull request by ganchoradkov**
Updated [CAIP-222](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-222.md) to support multi-chain authentication.

Added ability to 
- request signatures for multiple chains in a single request
- specify supported signing types via `signatureTypes` request parameter
- respond with a list of signed CACAOs

### CODE REVIEW

File: CAIPs/caip-222.md [hunk 1]
```diff
@@ -75,20 +77,30 @@ The JSON-RPC method is labelled as `wallet_authenticate` and expects the followi
 - statement (optional) - Human-readable ASCII assertion that the user will sign. It _MUST NOT_ contain `\n`.
 - requestId (optional) - System-specific identifier used to uniquely refer to the authentication request.
 - resources (optional) - List of information or references to information the user wishes to have resolved as part of the authentication by the relying party; express as [RFC 3986][rfc 3986] URIs and separated by `\n`.
+- signatureTypes (optional) - Object specifying the supported signing algorithms by the application for each namespace. The namespace MUST be defined in the key, while the supported signing algoriths as `string[]` and MUST contain one or more values if present.
+
+Example of `signatureTypes`
+
+```
+signatureTypes: {
+  "eip155": ["eip191", "eip1271" ],
```
Comment (oed) [inline CAIPs/caip-222.md:86]: So if I support both eip191 and eip1271 I can just not define this field and get either?

Comment (bumblefudge) [reply CAIPs/caip-222.md:86]: i think wildcards are kind of unsafe and likely to lead to bad UX deadends; better for dapps to have to explicitly list which kinds they support, I would think.

Comment (ganchoradkov) [reply CAIPs/caip-222.md:86]: The v1 (merged version) is very optimistic that the sender application would be able to validate the returned signature without knowing the signing type especially now that you could request non-evm signatures. This parameter lets you specify exactly which types you're supporting. Whether you can omit it, its up for debate. I've set it as optional to reduce devex friction.

File: CAIPs/caip-222.md [hunk 2]
```diff
@@ -45,7 +46,7 @@ The application would interface with a provider to make request as follows:
   "params": {
       "cacaov": string,
       "type": string,
-      "chainId": string,
+      "chains": string[],
```
Comment (arein) [inline CAIPs/caip-222.md:49]: should this be optional and the wallet can just respond with any of those it supports?

Comment (ganchoradkov) [reply CAIPs/caip-222.md:49]: The rationale is that the dapp would request signatures for all chains included in the `chains` parameter and the wallet would respond with signatures for those it supports as outlined [here](https://github.com/ChainAgnostic/CAIPs/blob/7087562ed1f3a49e62ca6eb739bc32b67961d1bc/CAIPs/caip-222.md#not-supported-chains)
>The wallet SHOULD ignore all unsupported chains and SHOULD NOT auto reject the request if there are supported chains to sign without explicit user rejection.

In v1, `chainId` is required parameter so it makes sense to preserve that to an extended.
Additionally, setting `chains` to explicitely empty `[]` array is valid and kinda allows to not set any chains already.

File: CAIPs/caip-222.md [hunk 3]
```diff
@@ -75,20 +77,30 @@ The JSON-RPC method is labelled as `wallet_authenticate` and expects the followi
 - statement (optional) - Human-readable ASCII assertion that the user will sign. It _MUST NOT_ contain `\n`.
 - requestId (optional) - System-specific identifier used to uniquely refer to the authentication request.
 - resources (optional) - List of information or references to information the user wishes to have resolved as part of the authentication by the relying party; express as [RFC 3986][rfc 3986] URIs and separated by `\n`.
+- signatureTypes (optional) - Object specifying the supported signing algorithms by the application for each namespace. The namespace MUST be defined in the key, while the supported signing algoriths as `string[]` and MUST contain one or more values if present.
```
Comment (ganchoradkov) [inline CAIPs/caip-222.md:80]:
```suggestion
- signatureTypes (optional) - Object specifying the supported signing algorithms by the application for each namespace. The namespace MUST be defined in the key, while the supported signing algorithms as `string[]` and MUST contain one or more values if present.
```
typo

**Comment by bumblefudge** [comment]
I'm still wondering if it makes more sense for the multi-CAIP222 to be CAIP-248 or CAIP-222; would love input from potential or early implementers.

**Comment by ganchoradkov** [comment]
> Member

@bumblefudge imho, it should be an update as it complely encapsulates the merged CAIP-222. Additionally, having them as different standards would cause conflics if one attempts to implement them at the same time in their current form forcing implementers to choose one OR the other

**Review by llbartekll — APPROVED** [review]

**Review by bumblefudge — APPROVED** [review]

**Pull request merged**